packages/adblocker/test/html-filtering.test.ts

```diff
@@ -212,6 +220,102 @@ describe('html-filtering', () => {
     });
+
+    describe('handles modifiers', () => {
+      it('handles simple forms', () => {
+        expect(
+          filter(
+            `{"trackingParam":"a"}`,
+            [],
+            [[new RegExp('"trackingParam":"(\\w+)"'), '"$1":""']],
+          ),
+        ).to.be.eql(`{"a":""}`);
+      });
+
+      it('handles html modifiers with global replaces', () => {
+        expect(filter(doc, [], [[new RegExp("__perfMark\\('.+?'\\);", 'g'), '']])).not.to.include(
+          "__perfMark('",
+        );
+        expect(filter(doc, [], [[new RegExp('redditstatic\\.com', 'g'), 'domain.tld']]))
+          .to.include('https://www.domain.tld/desktop2x/js/ads.js')
+          .to.include(
+            'https://www.domain.tld/desktop2x/RedesignContentFonts.509eef5d33306bd3b0d5.js',
+          )
+          .to.include(
+            'https://www.domain.tld/desktop2x/RedesignOldContentFonts.e450653685d17337cac6.js',
+          )
+          .not.to.include('redditstatic.com');
+      });
+
+      it('handles multiple modifiers', () => {
+        expect(
+          filter(
+            doc,
+            [],
+            [
+              [new RegExp('__SUPPORTS_TIMING_API &&'), 'false &&'],
+              [new RegExp('redditstatic\\.com', 'g'), 'domain.tld'],
+            ],
+          ),
+        )
+          .to.include('function __perfMark(name) { false && performance.mark(name); };')
+          .not.to.include('redditstatic.com');
+      });
+
+      it('handles html modifiers with html selectors', () => {
+        expect(
+          filter(
+            doc,
+            [['script', ["__perfMark('"]]],
+            [[new RegExp('(__firstLoaded = )false'), '$1true']],
+          ),
+        )
+          .not.to.include("__perfMark('")
+          .not.to.include('__firstLoaded = false');
+
+        // -- html selector should not be inferenced by modifiers
+        expect(filter(doc, [['script', ["__perfMark('"]]], [[new RegExp('script', 'g'), 'pre']]))
+          .not.to.include("__perfMark('")
+          .not.to.include('script');
+
+        expect(
+          filter(doc, [['script', ['app_html_start']]], [[new RegExp('app_html_start', 'g'), '']]),
+        ).not.to.include('app_html_start');
+      });
+
+      const urls = [
+        {
+          url: 'https://alliptvlinks.com/tktk-content/plugins/rwaKVtBAQPrF!vFGebtUlLymZ)E45/gzwGRql+kTV!Svj-ITOh+YiJ)Av-9.js',
```

Comment:
Maybe it be better for the reader to have an additional field for the path to the test fixture here instead of deriving it implicitly in `loadRequestSample`? Mostly, to have a link in both directions between the test and the fixture.

Perhaps:
```
{
  url: ...,
  filters: ...,
  fixture: "data/samples/alliptvlinks.com.br",
}
```

But just an idea.

Reply:
I split the function body into two, so we can consider this easier in future. [`4cc7ed0` (#3897)](https://github.com/ghostery/adblocker/pull/3897/commits/4cc7ed0558dc65ce1ad6245ce937242db61b0357)